src/model.ts

```diff
@@ -0,0 +1,194 @@
+    palette.palette[4].toStringHexRGB()
+  );
+}
+
+function set_inverse_layout_colors_light(palette: ColorPalette) {
+  document.body.style.setProperty(
+    '--jp-inverse-layout-color0',
+    palette.palette[0].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-inverse-layout-color1',
+    palette.palette[1].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-inverse-layout-color2',
+    palette.palette[2].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-inverse-layout-color3',
+    palette.palette[3].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-inverse-layout-color4',
+    palette.palette[4].toStringHexRGB()
+  );
+}
+
+/* function switch_text_to_black_or_white_light(
+  palette: ColorPalette,
+  threshold: number
+) {
+  const element_color = [];
+  const color = black;
+  for (let i = 0; i < palette.palette.length; i++) {
+    element_color[i] = set_black_or_white(palette.palette[i], color, threshold);
+  }
+
+  document.body.style.setProperty(
+    '--jp-ui-font-color0',
+    element_color[0].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-ui-font-color1',
+    element_color[1].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-ui-font-color2',
+    element_color[2].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-ui-font-color3',
+    element_color[3].toStringHexRGB()
+  );
+} */
+
+function set_ui_font_light(palette: ColorPalette) {
+  document.body.style.setProperty(
+    '--jp-ui-font-color0',
+    palette.palette[1].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-ui-font-color1',
+    palette.palette[2].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-ui-font-color2',
+    palette.palette[3].toStringHexRGB()
+  );
+  document.body.style.setProperty(
+    '--jp-ui-font-color3',
+    palette.palette[4].toStringHexRGB()
+  );
+}
+export class ThemeEditorLightModel extends VDomModel {
+  private _baseColor = '#FFFFFF';
+
+  public get baseColor(): string {
+    return this._baseColor;
+  }
+
+  public set baseColor(colorHEXstr: string) {
+    if (this._baseColor !== colorHEXstr) {
+      this._baseColor = colorHEXstr;
+      const colorValueRGBAstr = hexToRGBA(colorHEXstr);
+      this.stateChanged.emit();
+      const colorRGBA64 = new ColorRGBA64(
+        colorValueRGBAstr[0],
+        colorValueRGBAstr[1],
+        colorValueRGBAstr[2],
+        colorValueRGBAstr[3]
+      );
+
+      const palette: ColorPalette = new ColorPalette({
+        baseColor: colorRGBA64,
+        steps: 5,
+        interpolationMode: ColorInterpolationSpace.RGB
+      });
+      const colorHSL = rgbToHSL(colorRGBA64);
+      const h = colorHSL.h;
+      const s = colorHSL.s;
+      let l = 1 - colorHSL.l;
+
+      if (l > 0.2 && l <= 0.5) {
+        l = 0.05;
+      } else if (l > 0.5 && l <= 0.75) {
+        l = 0.95;
+      } else {
+        //pass
+      }
```

Comment:
Could you comment a bit on your choice here?

There is an inconsistency for `l < 0.2` and `l > 0.75` as it will take `l` when for `l == 0.3` for example that will result in taking `l = 0.05`. It sounds better to be consistent with the saturation (like using interval from ]0.05, 0.5] and ]0.5, 0.95[).

Reply:
This choice was a bit arbitrary, a bit empirical and of course not definitive. I will change this.